projectile.el

```diff
@@ -213,11 +213,21 @@ Otherwise consider the current directory the project root."
   :group 'projectile
   :type 'string)
 
-(defcustom projectile-build-dir "build"
-  "The directory Projectile will use for build systems that build out of tree."
+(defcustom projectile-compilation-dir "build"
+  "The directory Projectile will use when compiling out of tree.
+
+The directory path is relative to the project root.
+
+Overridable per project by setting `projectile-project-compilation-dir' in
+.dir-locals"
   :group 'projectile
   :type 'string)
 
+(defcustom projectile-prefer-out-of-tree-compilation t
```

Comment:
New defcustoms should all have a `:package-version` property. 

I don't like a lot the name of this `defcustom` and I wonder if it should be something like `projectile-compilation-style` with possible options `'in-project-tree` and `'outside-of-project-tree`. Not a big deal though.

Reply:
This only applies sometimes though. For CMake it's possible to both build in- and out-of-tree but for meson `'in-project-tree` is not allowed, and for all other `'outside-of-project-tree` is not allowed. So it really is a preference of sorts.